.github/actions/analyze/action.yml

```diff
@@ -58,6 +61,46 @@ inputs:
 runs:
   using: composite
   steps:
+
+  - name: Get latest commit hash from master
+    id: latest_commit
+    run: |
+      git fetch origin master
+      LATEST_COMMIT=$(git rev-parse origin/master)
+      echo "LATEST_COMMIT=$LATEST_COMMIT" >> $GITHUB_ENV
+    shell: bash
+
+  - name: Check if previous artifact exists
+    id: artifact_check
+    run: |
+      ARTIFACT_EXISTS=$(curl -s -H "Authorization: Bearer ${{ github.token }}" "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-${{ env.LATEST_COMMIT }}" | jq '.total_count' | awk '{print $1}')
+      if [[ $ARTIFACT_EXISTS -gt 0 ]]; then
+        echo "ARTIFACT_EXISTS=true" >> $GITHUB_ENV
+      else
+        echo "ARTIFACT_EXISTS=false" >> $GITHUB_ENV
+      fi
+    shell: bash
+
+  - name: Get artifact URL
+    id: get_artifact_url
+    run: |
+      ARTIFACT_URL=$(curl -s -H "Authorization: Bearer ${{ inputs.GITHUB_TOKEN }}" \
+        "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-${{ env.LATEST_COMMIT }}" | \
```

Comment:
```suggestion
        "https://api.github.com/repos/${{ github.repository }}/actions/artifacts?name=reports-$LATEST_COMMIT" | \
```

Reply:
This LATEST_COMMIT is taken from [this step](https://github.com/binary-com/deriv-app/pull/15172/files#diff-407accfedbc8dedafa97892cca68d7b11e1329aa1f8e836fb54967be01eb73faR70). I think `env.` is needed here.

Reply:
https://wikijs.deriv.cloud/en/Security/product-security/github-action-bestpractices#:~:text=When%20using%20Environment%20Variable%20directly%20use%20declared%20variable%20name%20(i.e%20%24SOMENAME)%20instead%20if%20interpolating%20it%20(i.e%20%24%7B%7B%20env.SOMENAME%20%7D%7D)%20as%20this%20is%20also%20vulnerable%20to%20Script%20Injections